js/src/lib/components/InferenceWidget/widgets/AudioClassificationWidget/AudioClassificationWidget.svelte

```diff
@@ -122,37 +122,36 @@
 		throw new TypeError("Invalid output: output must be of type Array<label: string, score:number>");
 	}
 
-	function applyInputSample(sample: WidgetExampleAssetInput<WidgetExampleOutputLabels>) {
-		file = null;
+	function applyInputSample(
+		sample: WidgetExampleAssetInput<WidgetExampleOutputLabels>,
+		{ isPreview = false, inferenceOpts = {} }: ExampleRunOpts = {}
+	) {
 		filename = sample.example_title!;
 		fileUrl = sample.src;
-		selectedSampleUrl = sample.src;
-		getOutput();
-	}
 
-	function previewInputSample(sample: WidgetExampleAssetInput<WidgetExampleOutputLabels>) {
-		filename = sample.example_title!;
-		fileUrl = sample.src;
-		if (isValidOutputLabels(sample.output)) {
-			output = sample.output;
-			outputJson = "";
-		} else {
-			output = [];
-			outputJson = "";
+		if (isPreview) {
+			if (isValidOutputLabels(sample.output)) {
+				output = sample.output;
+				outputJson = "";
+			} else {
+				output = [];
+				outputJson = "";
+			}
+			return;
 		}
+		file = null;
+		selectedSampleUrl = sample.src;
+		getOutput(inferenceOpts);
 	}
 
 	function validateExample(sample: WidgetExample): sample is WidgetExampleAssetInput<WidgetExampleOutputLabels> {
 		return isAssetInput(sample) && (!sample.output || isValidOutputLabels(sample.output));
 	}
 
 	onMount(() => {
-		const [exampleTitle, src] = getDemoInputs(model, ["example_title", "src"]);
-		if (callApiOnMount && src) {
-			filename = exampleTitle ?? "";
-			fileUrl = src;
-			selectedSampleUrl = src;
-			getOutput({ isOnLoadCall: true });
+		const example = getWidgetExample<WidgetExampleAssetInput<WidgetExampleOutputLabels>>(model, validateExample);
```

Comment:
I am wondering if this logic shouldn't live in `WidgetWrapper` instead?
Given it's the same for every widget

Reply:
I think putting this logic in [WidgetWrapper onMount](https://github.com/huggingface/hub-docs/blob/d2374e4e659bed1eb261df8b3eaae1e23c7d9789/js/src/lib/components/InferenceWidget/shared/WidgetWrapper/WidgetWrapper.svelte#L62) would be confusing. 
Here is the reason: all widgets except the [TextGeneration](https://github.com/huggingface/hub-docs/blob/aa2396eb94c5a23079fa5f528a6118a8f14ea4eb/js/src/lib/components/InferenceWidget/widgets/TextGenerationWidget/TextGenerationWidget.svelte#L66) & [Text2Image](https://github.com/huggingface/hub-docs/blob/85aac2e6505860cbebf6842a9d5e4a0b9f491638/js/src/lib/components/InferenceWidget/widgets/TextToImageWidget/TextToImageWidget.svelte#L46), have same code, which is:
```ts
if (callApiOnMount && example) {
	applyInputSample(example, { inferenceOpts: { isOnLoadCall: true } });
}
```
However, [TextGeneration](https://github.com/huggingface/hub-docs/blob/aa2396eb94c5a23079fa5f528a6118a8f14ea4eb/js/src/lib/components/InferenceWidget/widgets/TextGenerationWidget/TextGenerationWidget.svelte#L66) & [Text2Image](https://github.com/huggingface/hub-docs/blob/85aac2e6505860cbebf6842a9d5e4a0b9f491638/js/src/lib/components/InferenceWidget/widgets/TextToImageWidget/TextToImageWidget.svelte#L46) have custom logic. Which is:
```ts
const [textParam] = getSearchParams(["text"]);
if (textParam) {
	setTextAreaValue(textParam);
	getOutput({ useCache: true });
} else {
	const example = getWidgetExample<WidgetExampleTextInput<WidgetExampleOutputText>>(model, validateExample);
	if (example && callApiOnMount) {
		applyInputSample(example, { inferenceOpts: { isOnLoadCall: true, useCache: true } });
	}
}
```

Therefore, to implement it inside `WidgetWrapper onMount`, we would need to check if it is TextGen or Text2Image and getSearchParams, which seems too implicit & confusing. 
Wdyt?

Reply:
BTW is there some kind of Svelte component inheritance? does this concept exist?

Reply:
maybe @coyotte508 can also take a quick look and let us know if you have ideas on whether to factor that logic in WidgetWrapper or not?

Reply:
> I guess we can include the onMount call in a helper and have a useExample(model, callApiOnMount, validateExample) or useExample($$props) called here.

Thought about it. I think it actually makes it look quite confusing/unclean when you have to pass 2 callback functions: `validateExample` & `applyInputSample`

so it will look like this in that case:
`helpers.ts`
```ts
import { onMount } from "svelte"
...
export function customOnMount<T>(model: ModelData, isOnLoadCall: boolean, validate: func, applyInputSample: func){
    onMount(() => {
        const example = getExample<T>(model, validate);
        if(example && isOnLoadCall){
            applyInputSample(example);
        }
    })
}
```

and in `XyzWidget.svelte`
```ts
import { customOnMount } from "./helpers"
...
customOnMount<WidgetOutput<T>>(model, isOnLoadCall, validateExample, applyInputSample)
```

If you are fine, I'm happy to merge this PR as it is and iterate afterwards

Reply:
Made an update. 
All the onMount example runs are happening inside `WidgetWrapper`.
https://github.com/huggingface/hub-docs/blob/37f2ccdf6b8959342b2ec49e19f190cefdbd951d/js/src/lib/components/InferenceWidget/shared/WidgetWrapper/WidgetWrapper.svelte#L70

Specifically, there are two parts:
Part 1: run one of the examples from the model card
https://github.com/huggingface/hub-docs/blob/37f2ccdf6b8959342b2ec49e19f190cefdbd951d/js/src/lib/components/InferenceWidget/shared/WidgetWrapper/WidgetWrapper.svelte#L79-L82

Part 2: run examples from URL query param (for widgets like TextGen widget as mentioned [here](https://github.com/huggingface/hub-docs/pull/1023#discussion_r1360940977))
https://github.com/huggingface/hub-docs/blob/37f2ccdf6b8959342b2ec49e19f190cefdbd951d/js/src/lib/components/InferenceWidget/shared/WidgetWrapper/WidgetWrapper.svelte#L71-L77

To activate, Part 2, widget (example: TextGen widget) needs to pass/activate optional prop in WudgetWrapper
https://github.com/huggingface/hub-docs/blob/37f2ccdf6b8959342b2ec49e19f190cefdbd951d/js/src/lib/components/InferenceWidget/widgets/TextGenerationWidget/TextGenerationWidget.svelte#L198

@SBrandeis @coyotte508 wdyt?

Reply:
WDYT @SBrandeis ?